src/llama-context.cpp

```diff
@@ -1386,7 +1386,21 @@ void llama_context::output_reorder() {
 //
 
 uint32_t llama_context::graph_max_nodes() const {
-    return std::max<uint32_t>(1024u, 8u*model.n_tensors());
+    uint32_t base_nodes = std::max<uint32_t>(1024u, 8u*model.n_tensors());
```

Comment:
```suggestion
    uint32_t factor = LLM_ARCH_MEGREZ_MOE ? 9u : 8u;
    uint32_t base_nodes = std::max<uint32_t>(1024u, factor * model.n_tensors());
```

increase the `9u` if needed